finality-aleph/src/party.rs

```diff
@@ -258,20 +262,27 @@ where
             .runtime_api()
             .current_session(&BlockId::Number(prev_block_number))
         {
-            Ok(session) => session,
+            Ok(session) => {
+                self.sessions.insert(
+                    session_id,
+                    SessionInstance {
+                        session: session.clone(),
+                    },
+                );
+                session
+            }
             _ => {
                 error!(target: "afa", "No session found for current block #{}", 0);
                 return;
             }
         };
 
-        let current_stop_h = session.stop_h;
         let client = self.client.clone();
         let finalization_proposals_rx = &mut self.finalization_proposals_rx;
         let proposals_task = {
             let client = self.client.clone();
             async move {
-                while client.info().finalized_number < current_stop_h {
+                loop {
```

Comment:
I am not sure about this but it looks fine to just loop here

Reply:
We need `while` here. Otherwise, if we do not participate in the consensus (i.e. we are not an authority), we wouldn't end the task, which would prevent us from moving on to the next session